crypto/asn1/a_strex.c

```diff
@@ -454,10 +454,14 @@ static int do_name_ex(char_io *io_ch, void *arg, X509_NAME *n,
 
     cnt = X509_NAME_entry_count(n);
     for (i = 0; i < cnt; i++) {
+        /*
+         * Ugly casts to (X509_NAME *) ... but harmless 
+         * Waiting for the const'ification of STACK_OF api
+         */
         if (flags & XN_FLAG_DN_REV)
-            ent = X509_NAME_get_entry(n, cnt - i - 1);
+            ent = X509_NAME_get_entry((X509_NAME *)n, cnt - i - 1);
```

Comment:
Ping @mattcaswell : fixed in a new separated commit.
As stated, it is quite large ;)